database/pgsql/layer.go

```diff
@@ -329,6 +348,30 @@ func (pgSQL *pgSQL) InsertLayer(layer database.Layer) error {
 		}
 	}
 
+	// insert the layer's namespaces
+	stmt, err := tx.Prepare(insertLayerNamespace)
+
+	defer func() {
+		err = stmt.Close()
+		if err != nil {
+			tx.Rollback()
+			log.WithError(err).Error("failed to close prepared statement")
+		}
+	}()
+
+	if err != nil {
```

Comment:
move this above the defer or else the defer could possibly cause a nil ptr deference